src/data/sym2.lean

```diff
@@ -200,20 +215,32 @@ def diag (x : α) : sym2 α := ⟦(x, x)⟧
 /--
 A predicate for testing whether an element of `sym2 α` is on the diagonal.
 -/
-def is_diag (z : sym2 α) : Prop := z ∈ set.range (@diag α)
+def is_diag : sym2 α → Prop :=
+lift ⟨eq, λ _ _, propext eq_comm⟩
 
 @[simp]
-lemma diag_is_diag (a : α) : is_diag (diag a) :=
-by use a
+lemma is_diag_iff_eq {x y : α} : is_diag ⟦(x, y)⟧ ↔ x = y :=
+iff.rfl
 
 @[simp]
 lemma is_diag_iff_proj_eq (z : α × α) : is_diag ⟦z⟧ ↔ z.1 = z.2 :=
+prod.rec_on z $ λ _ _, is_diag_iff_eq
+
+@[simp]
+lemma diag_is_diag (a : α) : is_diag (diag a) :=
```

Comment:
```suggestion
lemma is_diag_diag (a : α) : is_diag (diag a) :=
```

Reply:
This name is the same as it was before this PR, but I suppose I could creep the scope to fix the name.

Reply:
I get confused about the name to give compositions like this, and I get even more confused when dot notation is involved.  If the conclusion were written as `(diag a).is_diag` (which it could be), would it still be `is_diag_diag`?

I think I'm guilty putting `is_foo` predicates at the ends of lemmas elsewhere, too, which is why I ask.

Reply:
I made [a Zulip thread](https://leanprover.zulipchat.com/#narrow/stream/113488-general/topic/.23naming.20and.20dot.20notation/near/246520191).

Reply:
I've also reverted the change - if it's contentious, then renaming it is outside the scope of this PR.